sunpy/map/mapbase.py

```diff
@@ -1482,6 +1484,12 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=4, scale=1.0,
         pad_x = int(np.max((diff[1], 0)))
         pad_y = int(np.max((diff[0], 0)))
 
+        if issubclass(self.data.dtype.type, numbers.Integral) and (np.isnan(missing)):
+            if version.parse(np.__version__) >= version.parse("1.20.0"):
+                raise ImportError('Missing NaN value is not supported for Integer map')
+            warnings.warn("Integer map data is incompatible with specified missing value",
+                          SunpyUserWarning)
```

Comment:
```suggestion
        if issubclass(self.data.dtype.type, numbers.Integral) and (missing % 1 != 0)):
            warnings.warn("The specified `missing` value is not an integer, but the data "
                          "array is of integer type, so the output may be strange.",
                          SunpyUserWarning)
```
- As noted in the discussion, we ought to raise a warning for all non-integer `missing`, not just NaN.
- Ditch the version check for NumPy.  It's going to error anyway from NumPy, and our warning message makes it clear why.  (Plus, the error shouldn't be `ImportError`.)